### PR TITLE
convert changelog and generate documentation in markdown format.

### PR DESCRIPTION
**Pull request by zhumin8**
[updated.]
- removed old `CHANGELOG.adoc` and replaced with `CHANGELOG.md`.
- in prep for markdown generation, added `first-page.adoc` and `configuration.adoc` (sections copied from index.adoc) so that result full doc can be concatenation of docs.
- produced doc per module to markdown to folder `docs/src/main/md/`
- in `spring-cloud-bus-pubsub.md`, replaced spring_cloud_bus_over_pubsub.png with link to github.

Used `asciidoctor` and `pandoc` to convert asciidoc to markdown.
Steps to generate the markdown files:

```
# convert changelog to md
sudo asciidoctor -b docbook CHANGELOG.adoc   
sudo pandoc -f docbook -t gfm CHANGELOG.xml -o CHANGELOG.md --shift-heading-level-by=1


# convert doc to md
cd docs/src/main/md
./convert-from-ascii.sh  

# manual copy title over from index.adoc (due to pandoc settings, this title is not carried over.)

```

### CODE REVIEW

File: docs/src/main/md/index.md [hunk 1]
```diff
@@ -0,0 +1,8780 @@
...
+to create the `DataSource`. You can select the type of connection pool
+(e.g., Tomcat, HikariCP, etc.) by [adding their dependency to the
+classpath](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-sql.html#boot-features-connect-to-production-database).
+
+Using the created `DataSource` in conjunction with Spring JDBC provides
+you with a fully configured and operational `JdbcTemplate` object that
+you can use to interact with your SQL database. You can connect to your
+database with as little as a database and instance names.
+
+## R2DBC Support
```
Comment (elefeint) [inline docs/src/main/md/index.md:750]: This needs to be indented one more level, so it's under Cloud SQL (this was also the problem in adoc; I just noticed it here).

File: docs/src/main/md/index.md [hunk 2]
```diff
@@ -0,0 +1,8780 @@
...
+The user properties and the properties provided by the
+`R2dbcCloudSqlEnvironmentPostProcessor` are then used by Spring Boot to
+create the `ConnectionFactory`.
+
+The customized `ConnectionFactory` is then ready to connect to Cloud
+SQL. The rest of Spring Data R2DBC objects built on it (
+`R2dbcEntityTemplate`, `DatabaseClient`) are automatically configured
+and operational, ready to interact with your SQL database.
+
+## Cloud SQL IAM database authentication
```
Comment (elefeint) [inline docs/src/main/md/index.md:852]: Also one more level of indentation.

File: docs/src/main/md/index.md [hunk 3]
```diff
@@ -0,0 +1,8780 @@
...
+2.  Add the IAM user or service account to the list of database users.
+
+3.  In the application settings, set
+    `spring.cloud.gcp.sql.enableIamAuth` to `true`. (Note that this will
+    also set the database protocol `sslmode` to `disabled`, as it’s
+    required for IAM authentication to work. However, it doesn’t
+    compromise the security of the communication because the connection
+    is always encrypted.)
+
+## Cloud SQL Configuration Properties
```
Comment (elefeint) [inline docs/src/main/md/index.md:872]: Also one more level of indentation.

File: docs/src/main/md/index.md [hunk 4]
```diff
@@ -0,0 +1,8780 @@
...
+
+``` xml
+<dependency>
+  <groupId>org.postgresql</groupId>
+  <artifactId>postgresql</artifactId>
+  <version>42.1.1</version>
+</dependency>
+```
+
+## Samples
```
Comment (elefeint) [inline docs/src/main/md/index.md:938]: Also one more level of indentation.

File: docs/src/main/md/index.md [hunk 5]
```diff
@@ -0,0 +1,8780 @@
...
+| `spring.cloud.gcp.sql.credentials.encoded-key`  | Base64-encoded contents of OAuth2 account private key in JSON format. Used to authenticate and authorize new connections to a Google Cloud SQL instance.                                                                                                                                                                                  | No       | Default credentials provided by the Spring GCP Boot starter |
+| `spring.cloud.gcp.sql.enableIamAuth`            | Specifies whether to enable IAM database authentication (PostgreSQL only).                                                                                                                                                                                                                                                                | No       | `False`                                                     |
+
+## Troubleshooting tips
```
Comment (elefeint) [inline docs/src/main/md/index.md:887]: Also one more level of indentation.

**Review by elefeint — APPROVED** [review]
LGTM, but I think we are going to find it harder to maintain the one single reference documentation file.

Is there any way to have individual per-product files that get combined into a single file with some pre/post processing when publishing?

**Comment by zhumin8** [comment]
this pr is rebased after #1164 merged in so that title level issues are addressed.
Based on the idea to have individual per-product files, revamped this pr to adding separate markdown docs. and leaving the concatenation of them as step in publishing (will be in`publish_javadoc11.sh` on separate pr).
Also updated pr description above to reflect latest changes.

**Pull request merged**